bug(openapi-generator): emits `label` on container fields (group/array/page), causing TS2322 in consumers

## Summary

The OpenAPI generator emits a `label` property on every generated field, including container types (`group`, `array`, `page`, `row`). These field types explicitly exclude `label` in their TypeScript definitions:

- `packages/dynamic-forms/src/lib/definitions/default/group-field.ts:22`: `readonly label?: never;`
- `packages/dynamic-forms/src/lib/definitions/default/array-field.ts:97`: `readonly label?: never;`
- Similar on `PageField` / `RowField`.

When a consumer imports the generated config and TypeScript type-checks it (via `as const satisfies FormConfig` at the end of the generated file), they see errors like:

```
error TS2322: Type '{ readonly key: "address"; readonly type: "group"; readonly label: "Address"; ... }' 
  is not assignable to type 'RegisteredFieldTypes'.
    Types of property 'label' are incompatible.
      Type '"Address"' is not assignable to type 'undefined'.
```

## Root cause

`packages/openapi-generator/src/mapper/schema-to-fields.ts` unconditionally sets `label` in two places:
- [Line 155](https://github.com/ng-forge/ng-forge/blob/main/packages/openapi-generator/src/mapper/schema-to-fields.ts#L155) — group handler (object schemas)
- [Line 204](https://github.com/ng-forge/ng-forge/blob/main/packages/openapi-generator/src/mapper/schema-to-fields.ts#L204) — general property handler (falls through for arrays as well)

The generator uses `schema.title ?? toLabel(prop.name)` — there's no filter based on the resolved `finalType`.

## Reproduction

Minimal OpenAPI 3.0 spec:

```yaml
openapi: '3.0.3'
info: { title: Repro, version: '1.0.0' }
paths:
  /entities:
    post:
      operationId: createEntity
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: object
              required: [id]
              properties:
                id: { type: string }
                address:
                  type: object
                  properties:
                    street: { type: string }
                tags:
                  type: array
                  items: { type: string }
      responses:
        '201': { description: Created }
```

Run `ng-forge-generator --spec repro.yaml --output out --interactive none`, then `tsc` on the generated file with `@ng-forge/dynamic-forms-material` augmentation active. Expected: clean compile. Actual: `TS2322` on every container field's `label`.

## Why this wasn't caught earlier

Integration tests in `packages/openapi-generator/src/__tests__/integration/` use string-grep on generated output, not a type-check. A compile-based test helper (`typecheckGeneratedForm`) was added in #344 that reveals this class of bug — running it against the new `nullable-edge-cases.yaml` fixture surfaces the `label` errors on groups and arrays cleanly. The existing `nested-objects.yaml` / `arrays.yaml` tests never got past the string-grep assertions.

## Proposed fix

In `mapPropertyToField` (and the matching path for nested groups), skip the `label` assignment when `finalType` is a container type. Rough sketch:

```ts
const CONTAINER_TYPES = new Set(['group', 'array', 'page', 'row']);

const field: FieldConfig = {
  key: prop.name,
  type: finalType,
};

// Only emit label for non-container field types
if (!CONTAINER_TYPES.has(finalType)) {
  field.label = (prop.schema as Record<string, unknown>)['title'] as string ?? toLabel(prop.name);
}
```

Same at [line 155](https://github.com/ng-forge/ng-forge/blob/main/packages/openapi-generator/src/mapper/schema-to-fields.ts#L155) for the nested-group handler.

## Acceptance criteria

- [ ] Generator does not emit `label` on `group`, `array`, `page`, or `row` fields.
- [ ] Existing `nested-objects.yaml` and `arrays.yaml` integration tests still pass (they currently don't assert on `label`, so should be unaffected).
- [ ] `nullable-edge-cases.yaml` (from #344) type-checks cleanly end-to-end.
- [ ] Consider adding the `typecheckGeneratedForm` helper to more fixtures as a follow-up to catch similar regressions.

## Notes

Discovered while investigating edge cases for #341 (nullable values). The nullable fix filters `nullable` on containers in the same spot — a similar filter for `label` fits the same pattern.

--- a/packages/openapi-generator/src/__tests__/integration/generate-pipeline.integration.spec.ts
+++ b/packages/openapi-generator/src/__tests__/integration/generate-pipeline.integration.spec.ts
@@ -176,6 +176,24 @@ describe('Nested Objects', () => {
     expect(form).toContain("key: 'city'");
     expect(form).toContain("key: 'zip'");
   });
+
+  it('should not emit label on group container fields (#348)', async () => {
+    await generate('nested-objects.yaml');
+
+    const form = await readGenerated(outputDir, 'forms', 'create-profile.form.ts');
+    // The address group should not carry a label (its children may).
+    const addressBlock = form.substring(form.indexOf("key: 'address'"), form.indexOf("key: 'street'"));
+    expect(addressBlock).not.toContain('label:');
+  });
+
+  it('should produce a form file that compiles against the published types (regression for #348)', async () => {
+    await generate('nested-objects.yaml');
+
+    const formPath = join(outputDir, 'forms', 'create-profile.form.ts');
+    const diagnostics = typecheckGeneratedForm(formPath);
+
+    expect(diagnostics, `Generated form should type-check cleanly. Errors:\n${diagnostics.join('\n')}`).toEqual([]);
+  }, 30_000); // tsc program creation + type-check is slow in CI
 });
 
 // ─── E. Arrays ───────────────────────────────────────────────────────
@@ -200,6 +218,27 @@ describe('Arrays', () => {
     const form = await readGenerated(outputDir, 'forms', 'create-team.form.ts');
     expect(form).toContain("key: 'scores'");
   });
+
+  it('should not emit label on array container fields (#348)', async () => {
+    await generate('arrays.yaml');
+
+    const form = await readGenerated(outputDir, 'forms', 'create-team.form.ts');
+    // Outer array fields should not carry a label; their templates / children may.
+    const membersBlock = form.substring(form.indexOf("key: 'members'"), form.indexOf('template:'));
+    expect(membersBlock).not.toContain('label:');
+    const scoresBlock = form.substring(form.indexOf("key: 'scores'"));
+    const scoresHeader = scoresBlock.substring(0, scoresBlock.indexOf('template:'));
+    expect(scoresHeader).not.toContain('label:');
+  });
+
+  it('should produce a form file that compiles against the published types (regression for #348)', async () => {
+    await generate('arrays.yaml');
+
+    const formPath = join(outputDir, 'forms', 'create-team.form.ts');
+    const diagnostics = typecheckGeneratedForm(formPath);
+
+    expect(diagnostics, `Generated form should type-check cleanly. Errors:\n${diagnostics.join('\n')}`).toEqual([]);
+  }, 30_000); // tsc program creation + type-check is slow in CI
 });
 
 // ─── F. allOf ────────────────────────────────────────────────────────

--- a/packages/openapi-generator/src/__tests__/integration/generate-pipeline.integration.spec.ts
+++ b/packages/openapi-generator/src/__tests__/integration/generate-pipeline.integration.spec.ts
@@ -181,9 +181,10 @@ describe('Nested Objects', () => {
     await generate('nested-objects.yaml');
 
     const form = await readGenerated(outputDir, 'forms', 'create-profile.form.ts');
-    // The address group should not carry a label (its children may).
-    const addressBlock = form.substring(form.indexOf("key: 'address'"), form.indexOf("key: 'street'"));
-    expect(addressBlock).not.toContain('label:');
+    // Anchored regex avoids the silently-passing-empty-substring trap if codegen
+    // ever reorders properties: a `label:` line MUST NOT appear between the
+    // `key: 'address'` and `type: 'group'` lines (with any whitespace in between).
+    expect(form).toMatch(/key: 'address',\s*type: 'group',\s*(?!label:)/);
   });
 
   it('should produce a form file that compiles against the published types (regression for #348)', async () => {
@@ -223,12 +224,10 @@ describe('Arrays', () => {
     await generate('arrays.yaml');
 
     const form = await readGenerated(outputDir, 'forms', 'create-team.form.ts');
-    // Outer array fields should not carry a label; their templates / children may.
-    const membersBlock = form.substring(form.indexOf("key: 'members'"), form.indexOf('template:'));
-    expect(membersBlock).not.toContain('label:');
-    const scoresBlock = form.substring(form.indexOf("key: 'scores'"));
-    const scoresHeader = scoresBlock.substring(0, scoresBlock.indexOf('template:'));
-    expect(scoresHeader).not.toContain('label:');
+    // Anchored regexes: outer array fields must not carry a `label:` between
+    // their `key`/`type` header (templates and child fields are allowed to).
+    expect(form).toMatch(/key: 'members',\s*type: 'array',\s*(?!label:)/);
+    expect(form).toMatch(/key: 'scores',\s*type: 'array',\s*(?!label:)/);
   });
 
   it('should produce a form file that compiles against the published types (regression for #348)', async () => {

--- a/packages/openapi-generator/src/generator/form-config-generator.spec.ts
+++ b/packages/openapi-generator/src/generator/form-config-generator.spec.ts
@@ -397,4 +397,18 @@ describe('array template rendering', () => {
     expect(result).not.toContain('addButton');
     expect(result).not.toContain('removeButton');
   });
+
+  it('should omit label line when label is undefined (#348)', () => {
+    const fields: FieldConfig[] = [
+      {
+        key: 'address',
+        type: 'group',
+        fields: [{ key: 'street', type: 'input', label: 'Street' }],
+      },
+    ];
+
+    const result = generateFormConfig(fields, defaultOptions);
+    const addressBlock = result.substring(result.indexOf("key: 'address'"), result.indexOf("key: 'street'"));
+    expect(addressBlock).not.toContain('label:');
+  });
 });

--- a/packages/openapi-generator/src/generator/form-config-generator.ts
+++ b/packages/openapi-generator/src/generator/form-config-generator.ts
@@ -35,7 +35,9 @@ function generateFieldLines(field: FieldConfig, indent: string): string[] {
   lines.push(`${indent}{`);
   lines.push(`${indent}  key: '${escapeString(field.key)}',`);
   lines.push(`${indent}  type: '${escapeString(field.type)}',`);
-  lines.push(`${indent}  label: '${escapeString(field.label)}',`);
+  if (field.label !== undefined) {
+    lines.push(`${indent}  label: '${escapeString(field.label)}',`);
+  }
 
   if (field.value !== undefined) {
     lines.push(`${indent}  value: ${JSON.stringify(field.value)},`);

--- a/packages/openapi-generator/src/mapper/schema-to-fields.spec.ts
+++ b/packages/openapi-generator/src/mapper/schema-to-fields.spec.ts
@@ -596,12 +596,13 @@ describe('mapSchemaToFields', () => {
     // First field: discriminator radio
     expect(result.fields[0]).toMatchObject({ key: 'petType', type: 'radio' });
 
-    // Second field: dog variant group with logic
+    // Second field: dog variant group with logic.
+    // Group fields declare `label?: never` (issue #348), so no label is emitted.
     expect(result.fields[1]).toMatchObject({
       key: 'dogVariant',
       type: 'group',
-      label: 'Dog',
     });
+    expect(result.fields[1].label).toBeUndefined();
     expect(result.fields[1].logic).toEqual([
       {
         type: 'hidden',
@@ -616,12 +617,13 @@ describe('mapSchemaToFields', () => {
     expect(result.fields[1].fields).toHaveLength(1);
     expect(result.fields[1].fields![0].key).toBe('breed');
 
-    // Third field: cat variant group with logic
+    // Third field: cat variant group with logic.
+    // Group fields declare `label?: never` (issue #348), so no label is emitted.
     expect(result.fields[2]).toMatchObject({
       key: 'catVariant',
       type: 'group',
-      label: 'Cat',
     });
+    expect(result.fields[2].label).toBeUndefined();
     expect(result.fields[2].logic).toEqual([
       {
         type: 'hidden',
@@ -1036,6 +1038,104 @@ describe('mapSchemaToFields', () => {
       const result = mapSchemaToFields(schema, []);
       // 'name' has no match, 'telephoneDescription' ends in 'Description' not 'telephone'
       expect(result.fields[0].props?.['type']).toBe('text');
+    });
+  });
+
+  // Regression tests for #348: container types declare `label?: never` in their TS
+  // definitions, so emitting a label causes TS2322 in consumers.
+  describe('container types: no label emitted (#348)', () => {
+    it('should not emit label on group fields', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          address: {
+            type: 'object',
+            title: 'Address',
+            properties: {
+              street: { type: 'string' },
+            },
+          } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('group');
+      expect(result.fields[0].label).toBeUndefined();
+    });
+
+    it('should not emit label on array fields (object items)', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          contacts: {
+            type: 'array',
+            title: 'Contacts',
+            items: {
+              type: 'object',
+              properties: { email: { type: 'string' } },
+            },
+          } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('array');
+      expect(result.fields[0].label).toBeUndefined();
+    });
+
+    it('should not emit label on array fields (primitive items)', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          tags: {
+            type: 'array',
+            title: 'Tags',
+            items: { type: 'string' },
+          } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('array');
+      expect(result.fields[0].label).toBeUndefined();
+    });
+
+    it('should not emit label on group fields produced by oneOf+discriminator nested schemas', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          payment: {
+            title: 'Payment',
+            discriminator: { propertyName: 'kind' },
+            oneOf: [
+              {
+                type: 'object',
+                properties: {
+                  kind: { type: 'string', enum: ['card'] },
+                  number: { type: 'string' },
+                },
+              } as SchemaObject,
+            ],
+          } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('group');
+      expect(result.fields[0].label).toBeUndefined();
+    });
+
+    it('should still emit label on value-bearing fields', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          email: { type: 'string', title: 'Email Address' } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('input');
+      expect(result.fields[0].label).toBe('Email Address');
     });
   });
 });

--- a/packages/openapi-generator/src/mapper/schema-to-fields.spec.ts
+++ b/packages/openapi-generator/src/mapper/schema-to-fields.spec.ts
@@ -598,6 +598,8 @@ describe('mapSchemaToFields', () => {
 
     // Second field: dog variant group with logic.
     // Group fields declare `label?: never` (issue #348), so no label is emitted.
+    // Note: `toMatchObject` would silently pass if a label leaked back in — keep
+    // the explicit `toBeUndefined()` assertion below.
     expect(result.fields[1]).toMatchObject({
       key: 'dogVariant',
       type: 'group',
@@ -619,6 +621,8 @@ describe('mapSchemaToFields', () => {
 
     // Third field: cat variant group with logic.
     // Group fields declare `label?: never` (issue #348), so no label is emitted.
+    // Note: `toMatchObject` would silently pass if a label leaked back in — keep
+    // the explicit `toBeUndefined()` assertion below.
     expect(result.fields[2]).toMatchObject({
       key: 'catVariant',
       type: 'group',
@@ -1136,6 +1140,19 @@ describe('mapSchemaToFields', () => {
       const result = mapSchemaToFields(schema, []);
       expect(result.fields[0].type).toBe('input');
       expect(result.fields[0].label).toBe('Email Address');
+    });
+
+    it('should not emit label when x-ng-forge-type overrides to a container type', () => {
+      const schema: SchemaObject = {
+        type: 'object',
+        properties: {
+          custom: { type: 'string', title: 'Custom', 'x-ng-forge-type': 'container' } as unknown as SchemaObject,
+        },
+      };
+
+      const result = mapSchemaToFields(schema, []);
+      expect(result.fields[0].type).toBe('container');
+      expect(result.fields[0].label).toBeUndefined();
     });
   });
 });

--- a/packages/openapi-generator/src/mapper/schema-to-fields.ts
+++ b/packages/openapi-generator/src/mapper/schema-to-fields.ts
@@ -19,8 +19,12 @@ const NULLABLE_SUPPORTED_FIELD_TYPES = new Set(['input', 'textarea', 'select', '
  * Field types whose runtime definition declares `label?: never` and therefore
  * rejects any `label` assignment under `as const satisfies FormConfig`. Emitting
  * a label on these types causes TS2322 in consumers (issue #348).
+ *
+ * `container` is included defensively: while the schema mapper never produces it
+ * directly, an `x-ng-forge-type: container` override can route through this code
+ * path, and `BaseContainerField` declares the same `label?: never` constraint.
  */
-const CONTAINER_FIELD_TYPES = new Set(['group', 'array', 'row', 'page']);
+const CONTAINER_FIELD_TYPES = new Set(['group', 'array', 'row', 'page', 'container']);
 
 function singularize(label: string): string {
   // Only strip trailing 's' if word is 4+ chars and doesn't end in 'ss'
@@ -339,8 +343,13 @@ function mapPropertyToField(
         const templateField: FieldConfig = {
           key: 'value',
           type: itemTypeResult.fieldType,
-          label: fieldLabel,
         };
+        // Container types declare `label?: never` (issue #348). `itemTypeResult.fieldType`
+        // is never a container today, but gate symmetrically with the property handler
+        // so future routing changes can't slip through.
+        if (!CONTAINER_FIELD_TYPES.has(itemTypeResult.fieldType)) {
+          templateField.label = fieldLabel;
+        }
         if (itemTypeResult.props && Object.keys(itemTypeResult.props).length > 0) {
           templateField.props = itemTypeResult.props;
         }

--- a/packages/openapi-generator/src/mapper/schema-to-fields.ts
+++ b/packages/openapi-generator/src/mapper/schema-to-fields.ts
@@ -15,6 +15,13 @@ import { logger } from '../utils/logger.js';
  */
 const NULLABLE_SUPPORTED_FIELD_TYPES = new Set(['input', 'textarea', 'select', 'radio', 'multi-checkbox', 'slider', 'datepicker']);
 
+/**
+ * Field types whose runtime definition declares `label?: never` and therefore
+ * rejects any `label` assignment under `as const satisfies FormConfig`. Emitting
+ * a label on these types causes TS2322 in consumers (issue #348).
+ */
+const CONTAINER_FIELD_TYPES = new Set(['group', 'array', 'row', 'page']);
+
 function singularize(label: string): string {
   // Only strip trailing 's' if word is 4+ chars and doesn't end in 'ss'
   if (label.length >= 4 && label.endsWith('s') && !label.endsWith('ss')) {
@@ -31,7 +38,7 @@ export interface LogicConfig {
 export interface FieldConfig {
   key: string;
   type: string;
-  label: string;
+  label?: string;
   value?: unknown;
   nullable?: boolean;
   placeholder?: string;
@@ -87,12 +94,12 @@ export function mapSchemaToFields(schema: SchemaObject, requiredFields: string[]
         ambiguousFields.push(...variantResult.ambiguousFields);
         warnings.push(...variantResult.warnings);
 
-        // Wrap variant fields in a group with logic for conditional visibility
+        // Wrap variant fields in a group with logic for conditional visibility.
+        // Group fields declare `label?: never` (issue #348), so no label is emitted.
         if (variantFields.length > 0) {
           fields.push({
             key: `${group.discriminatorValue}Variant`,
             type: 'group',
-            label: toEnumLabel(group.discriminatorValue),
             fields: variantFields,
             logic: [
               {
@@ -149,10 +156,10 @@ function mapPropertyToField(
     ambiguousFields.push(...innerResult.ambiguousFields);
     warnings.push(...innerResult.warnings);
 
+    // Group fields declare `label?: never` (issue #348), so no label is emitted.
     return {
       key: prop.name,
       type: 'group',
-      label: ((prop.schema as Record<string, unknown>)['title'] as string) ?? toLabel(prop.name),
       fields: innerResult.fields,
     };
   }
@@ -201,8 +208,12 @@ function mapPropertyToField(
   const field: FieldConfig = {
     key: prop.name,
     type: finalType,
-    label: ((prop.schema as Record<string, unknown>)['title'] as string) ?? toLabel(prop.name),
   };
+
+  // Container types declare `label?: never` (issue #348) — only emit on value-bearing types.
+  if (!CONTAINER_FIELD_TYPES.has(finalType)) {
+    field.label = ((prop.schema as Record<string, unknown>)['title'] as string) ?? toLabel(prop.name);
+  }
 
   // readOnly → disabled
   if ((prop.schema as Record<string, unknown>)['readOnly']) {

